Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -870,10 +870,13 @@ Ref<Scrollbar> RenderLayerScrollableArea::createScrollbar(ScrollbarOrientation o
     auto& actualRenderer = *rendererForScrollbar(renderer);
     bool hasCustomScrollbarStyle = is<RenderBox>(actualRenderer) && downcast<RenderBox>(actualRenderer).style().hasPseudoStyle(PseudoId::Scrollbar);
     auto element = downcast<RenderBox>(actualRenderer).element();
-    if (hasCustomScrollbarStyle && element)
+    if ((hasCustomScrollbarStyle || scrollbarWidth() == ScrollbarWidth::None) && element)
```

Comment:
I think ideally this would just go down the native scrollbar code path? And then hide it inside of Scrollbar.cpp but when I tried this I couldn't get it working. It would still reserve space and show when stationary but flicker away on scroll.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=249278 ?

Reply:
That bug definitely sounds related to this issue.

Reply:
I think in this case the issue may have been to do with the fact that the scrollbar thickness was being incorrectly set to full width and this messed with the rendering despite nativeScrollbarVisibility being set to HiddenByStyle.

Will hopefully be able to fully implement it using the native codepath when I make the new PR that implements the code (I'm splitting this one up)